### PR TITLE
Add Content for Astro Audit Logs

### LINKED ISSUE #1253: Document Audit Logs
### Context

From @dnarain-astro:

- Audit Logs are starting to take shape and are expected to ship in October. We need to document them.
- PB card here: https://astronomer.productboard.com/feature-board/2193567-step-1-feature-backlog/features/9156095/detail

### Ask
Figure out how to document audit logs from:
- End user perspective
- Technical reference material on the format of the logs (I don’t think we have an obvious place to put this in the current docs structure)

Inspiration: https://docs.cloudera.com/management-console/cloud/auditing/topics/mc-cp-auditing-data-model.html

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #1253

### CODE REVIEW

File: astro/view-logs.md [hunk 1]
```diff
@@ -71,3 +71,90 @@ Astro also supports exporting Airflow metrics to Datadog. See [Export Airflow me
 1. Create a new Datadog API key or copy an existing API key. See [API and Application Keys](https://docs.datadoghq.com/account_management/api-app-keys/).
 2. Identify the Astro cluster from which you want to forward task logs.
 3. Submit a request to [Astronomer support](https://cloud.astronomer.io/support) with your Datadog API key, the name of your Astro cluster, and the [Datadog Site](https://docs.datadoghq.com/getting_started/site/) where you want the logs forwarded.
+
+## Astro audit logs
+
+Astro audit logs record Astro control plane administrative activities and events. You can use the audit logs to determine who did what, where, and when. You can also use audit logs to ensure your organization is meeting its security and regulatory requirements. For example, you can use audit logs to prove that your organization met the Center for Internet Security (CIS) password security requirements for a specific period.
+
```
Comment (dnarain-astro) [inline astro/view-logs.md:78]: We can't support this specific example (CIS password security) in Audit Logs, but here are some other examples of things we can do:
"track user assignments to your organization or to individual workspaces"
"track each time a user accesses the Airflow UI"
"track each time new code was deployed to an environment"

File: astro/view-logs.md [hunk 2]
```diff
@@ -71,3 +71,90 @@ Astro also supports exporting Airflow metrics to Datadog. See [Export Airflow me
...
+The audit log file is provided as a JSON file. Every entry in the audit log file corresponds to an event, and event attributes provide additional information about the specific event. 
+
+#### Common fields
+
+The following table lists the common fields shared by all three categories of event data.
+
+| Field  | Description                                          |
+| ------- | ---------------------------------------------------- |
+| `subjectId` | A unique identifier that identifies the initiator of the request. The value can be USER, or an API key. |
+| `subjectType` | A unique identifier that identifies the subject of the request. The value can be USER, or a service key. |
```
Comment (dnarain-astro) [inline astro/view-logs.md:96]: its literally either USER or SERVICEKEY - i.e. not the service key that was used, but that the type of the subjectId is SERVICEKEY.

File: astro/view-logs.md [hunk 3]
```diff
@@ -71,3 +71,90 @@ Astro also supports exporting Airflow metrics to Datadog. See [Export Airflow me
...
+
+The audit log file is provided as a JSON file. Every entry in the audit log file corresponds to an event, and event attributes provide additional information about the specific event. 
+
+#### Common fields
+
+The following table lists the common fields shared by all three categories of event data.
+
+| Field  | Description                                          |
+| ------- | ---------------------------------------------------- |
+| `subjectId` | A unique identifier that identifies the initiator of the request. The value can be USER, or an API key. |
```
Comment (dnarain-astro) [inline astro/view-logs.md:95]: Here, the value is the Id of the user or the API key (i.e. its not the literal USER - that is in the subjectType, but the unique id of the user - e.g. clhiudhfioaewh52"

File: astro/view-logs.md [hunk 4]
```diff
@@ -71,3 +71,90 @@ Astro also supports exporting Airflow metrics to Datadog. See [Export Airflow me
...
+
+| Field  | Description                                          |
+| ------- | ---------------------------------------------------- |
+| `operationName` | The name of the API event. For example, `createUserInvite` or `workspaceCreate`. |
+| `graphqlClientName` | The type of client making the request. The values are `cloud-ui` or `cli`. |
+| `requestInput` | The input for the API request. |
+| `requestBody` | Raw graphQL for the event. |
+
+The following table maps some common `operationName` attributes to their corresponding `requestInput` attributes. 
+
```
Comment (dnarain-astro) [inline astro/view-logs.md:115]: Should we clarify that this is not an exhaustive list?  The log goes on for gigabytes and is full of action - I just picked the ones that are most interesting and listed them.

File: astro/view-logs.md [hunk 5]
```diff
@@ -71,3 +71,90 @@ Astro also supports exporting Airflow metrics to Datadog. See [Export Airflow me
...
+Astro audit logs record Astro control plane administrative activities and events. You can use the audit logs to determine who did what, where, and when. You can also use audit logs to ensure your organization is meeting its security and regulatory requirements. For example, you can use audit logs to prove that your organization met the Center for Internet Security (CIS) password security requirements for a specific period.
+
+### Available audit log data
+
+The following categories of event data are available in the audit log:
+
+- API events: Data generated by user actions in the Cloud UI, Astro CLI, or from internal processing actions in the Astro control plane.
+- Airflow UI access: Data generated when users access the Airflow UI.
+- Astronomer container registry access: Data generated when users access the Astronomer container registry through the control plane authentication policy.
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply astro/view-logs.md:85]: @dnarain-astro I have updated the text as recommended.

File: astro/view-logs.md (hunk 1 above)
Comment (kwcanuck) [reply astro/view-logs.md:78]: @dnarain-astro I have updated the text.

File: astro/view-logs.md (hunk 3 above)
Comment (kwcanuck) [reply astro/view-logs.md:95]: @dnarain-astro the edit has been implemented.

File: astro/view-logs.md (hunk 4 above)
Comment (kwcanuck) [reply astro/view-logs.md:115]: @dnarain-astro as discussed, I have added a statement below the table indicating how additional mappings can be located. In the introductory statement for the table, I indicate that they are some of the most common mappings. Users should understand that the list is not comprehensive.

**Review by dnarain-astro — APPROVED** [review]
Thank You! LGTM

**Review by maxdml — APPROVED** [review]

**Comment by kwcanuck** [comment]
@jwitz this PR is ready for your review.

File: astro/view-logs.md [hunk 6]
```diff
@@ -71,3 +71,92 @@ Astro also supports exporting Airflow metrics to Datadog. See [Export Airflow me
 1. Create a new Datadog API key or copy an existing API key. See [API and Application Keys](https://docs.datadoghq.com/account_management/api-app-keys/).
 2. Identify the Astro cluster from which you want to forward task logs.
 3. Submit a request to [Astronomer support](https://cloud.astronomer.io/support) with your Datadog API key, the name of your Astro cluster, and the [Datadog Site](https://docs.datadoghq.com/getting_started/site/) where you want the logs forwarded.
+
+## Astro audit logs
+
+Astro audit logs record Astro control plane administrative activities and events. You can use the audit logs to determine who did what, where, and when. You can also use audit logs to ensure your organization is meeting its security and regulatory requirements. For example, you can use audit logs to determine when users were added to your organization or to individual Workspaces.
```
Comment (jwitz) [inline astro/view-logs.md:77]: Great intro!

File: astro/view-logs.md [hunk 7]
```diff
@@ -71,3 +71,92 @@ Astro also supports exporting Airflow metrics to Datadog. See [Export Airflow me
...
+
+### Available audit log data
+
+The following categories of event data are available in the audit log:
+
+- API events: Data is generated by user actions in the Cloud UI, Astro CLI, or from internal processing actions in the Astro control plane.
```
Comment (jwitz) [inline astro/view-logs.md:83]:
```suggestion
- API events: The data generated by user actions in the Cloud UI, Astro CLI, or from internal processing actions in the Astro control plane.
```

File: astro/view-logs.md [hunk 8]
```diff
@@ -71,3 +71,92 @@ Astro also supports exporting Airflow metrics to Datadog. See [Export Airflow me
...
+The audit log file is provided as a JSON file. Every entry in the audit log file corresponds to an event, and event attributes provide additional information about the specific event. 
+
+#### Common fields
+
+The following table lists the common fields shared by all three categories of event data.
+
+| Field  | Description                                          |
+| ------- | ---------------------------------------------------- |
+| `subjectId` | A unique identifier that identifies the initiator of the request. The value can be user ID or an API key. |
+| `subjectType` | A unique identifier that identifies the subject of the request. The value can be USER or SERVICEKEY. |
```
Comment (jwitz) [inline astro/view-logs.md:96]:
```suggestion
| `subjectType` | A unique identifier that identifies the subject of the request. The value can be `USER` or `SERVICEKEY`. |
```

File: astro/view-logs.md [hunk 9]
```diff
@@ -71,3 +71,92 @@ Astro also supports exporting Airflow metrics to Datadog. See [Export Airflow me
...
+The following table lists the fields that are unique to API events.
+
+| Field  | Description                                          |
+| ------- | ---------------------------------------------------- |
+| `operationName` | The name of the API event. For example, `createUserInvite` or `workspaceCreate`. |
+| `graphqlClientName` | The type of client making the request. The values are `cloud-ui` or `cli`. |
+| `requestInput` | The input for the API request. |
+| `requestBody` | Raw graphQL for the event. |
+
+The following table maps some common `operationName` attributes to their corresponding `requestInput` attributes.
```
Comment (jwitz) [inline astro/view-logs.md:114]: Does this need a title?

**Review by jwitz — APPROVED** [review]
Looks great @kwcanuck ! Adding the blocked label until the feature ships.

File: astro/view-logs.md (hunk 9 above)
Comment (kwcanuck) [reply astro/view-logs.md:114]: @jwitz a title isn't required as this information is part of the API event fields.

File: astro/view-logs.md [hunk 10]
```diff
@@ -71,3 +71,92 @@ Astro also supports exporting Airflow metrics to Datadog. See [Export Airflow me
...
+
+The following table lists the fields that are unique to Astronomer container registry access events.
+
+| Field  | Description                                          |
+| ------- | ---------------------------------------------------- |
+| `deploymentId` | A unique identifier that identifies the Deployment on which the event occurred. |
+| `path` | The path to the image in the registry. |
+| `method` | The HTTP method for the request. |
+
+### Export audit logs
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply astro/view-logs.md:153]: @jwitz - typically, conceptual and reference topics precede tasks within a section. This is the correct location for this information.

File: astro/view-logs.md [hunk 11]
```diff
@@ -69,5 +69,3 @@ Astro supports forwarding Airflow task logs to [Datadog](https://www.datadoghq.c
 1. Create a new Datadog API key or copy an existing API key. See [API and Application Keys](https://docs.datadoghq.com/account_management/api-app-keys/).
 2. Identify the Astro cluster from which you want to forward task logs.
 3. Submit a request to [Astronomer support](https://cloud.astronomer.io/support) with your Datadog API key, the name of your Astro cluster, and the [Datadog Site](https://docs.datadoghq.com/getting_started/site/) where you want the logs forwarded.
```
Comment (jwitz) [inline astro/view-logs.md:71]:
```suggestion
3. Submit a request to [Astronomer support](https://cloud.astronomer.io/support) with your Datadog API key, the name of your Astro cluster, and the [Datadog Site](https://docs.datadoghq.com/getting_started/site/) where you want the logs forwarded.

Astro also supports exporting Airflow metrics to Datadog. See [Export Airflow metrics to Datadog](deployment-metrics.md#export-airflow-metrics-to-datadog).
```

**Review by jwitz — APPROVED** [review]

File: static/robots.txt [hunk 12]
```diff
@@ -7,6 +7,7 @@ Disallow: /software/
 Disallow: /software_versioned_sidebars/
 Disallow: /README.md
 Disallow: /docusaurus.config.js
+Disallow: /astro/audit-logs.md
```
Comment (jwitz) [inline static/robots.txt:10]:
```suggestion
Disallow: /astro/astro-audit-logs.md
```

**Pull request merged**